From: Thomas Renninger <trenn@suse.de>
Subject: Do not load acpi_cpufreq unconditionally
Patch-mainline: yes
References: 218377 

The acpi-cpufreq module currently even loads on machines that
do not support cpufreq and break suspend and possible other things
there. This patch goes back to behaviour to let the module only load
if ACPI subsystem provides cpufreq data.

Signed-off-by: Thomas Renninger <trenn@suse.de>
Signed-off-by: Dave Jones <davej@redhat.com>

--- x/arch/i386/kernel/cpu/cpufreq/acpi-cpufreq.c
+++ y/arch/i386/kernel/cpu/cpufreq/acpi-cpufreq.c
@@ -597,7 +597,6 @@ static struct cpufreq_driver acpi_cpufre
 	.name	= "acpi-cpufreq",
 	.owner	= THIS_MODULE,
 	.attr	= acpi_cpufreq_attr,
-	.flags	= CPUFREQ_STICKY,
 };
 
 
@@ -608,7 +607,7 @@ acpi_cpufreq_init (void)
 
 	acpi_cpufreq_early_init_acpi();
 
- 	return cpufreq_register_driver(&acpi_cpufreq_driver);
+	return cpufreq_register_driver(&acpi_cpufreq_driver);
 }
 
 
